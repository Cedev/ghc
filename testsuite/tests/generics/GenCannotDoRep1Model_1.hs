{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE TypeOperators #-}
{-# LANGUAGE PolyKinds #-}

module CannotDoRep1Model_2 where

import GHC.Generics (Generic, Generic1)


-- We cannot generate a useful generic representation for this strange composition

data Strange f = Strange (f (f Int))                   deriving Generic1
