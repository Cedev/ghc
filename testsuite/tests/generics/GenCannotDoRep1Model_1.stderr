
GenCannotDoRep1Model_1.hs:12:65: error:
    • Can't make a derived instance of ‘Generic1 Strange’:
        Constructor ‘Strange’ applies the parameter but the application
                              isn't of the the form (p (f p)) or (p c)
    • In the data declaration for ‘Strange’
